src/frontend/state/LDProvider.tsx

```diff
@@ -0,0 +1,46 @@
+import React, { useEffect, useState } from 'react'
+import { asyncWithLDProvider } from 'launchdarkly-react-client-sdk'
+import { LDEnv } from 'common/types'
+
+type Props = {
+  children: React.ReactNode
+}
+
+const LDProvider = ({ children }: Props): React.ReactElement => {
+  const [ldConfig, setLDConfig] = useState<LDEnv>({
+    envId: '',
+    ldUser: {
+      kind: 'user',
+      key: ''
+    }
+  })
+  const [provider, setProvider] = useState<React.ReactElement>(<></>)
```

Comment:
Have you tried what happens if we try to use `useFlags ` or `useLDClient` before the provider is initialized? We could get some errors because those hooks are expected to be used inside of an `<LDProvider />` but they initially are not.

Reply:
wdym? the LDProvider is being initialized with the App here:
https://github.com/HyperPlay-Gaming/hyperplay-desktop-client/pull/674/files#diff-39a743b1ad7d31985313996d6fa54533b97d5ee693be0f992aca8ca2f653e7e4R114

So in theory, anything that uses these hooks on the app should work.

Reply:
Another approach would be to use the JS SDK and load it on the head of the index.html, but we look the hooks.

Reply:
Oh no need, I feel we're on the right track with using hooks. 

What I mean is that from what I see`<LDProvider />` is initially `<></>` but after we load the config it becomes `<AsyncLDProvider/>`. In that period maybe using LD hooks could throw an error because they're not wrapped yet in the provider they need.

Either way I have not run the code locally so might be wrong.